### ISSUE TITLE
⬆️ Dokka 1.8.10

### ISSUE THREAD

**Issue by LVMVRQUXL** [kotools/types#285]
## 📝 Description

We would like to upgrade the Dokka Gradle plugin from [1.7.20](https://github.com/Kotlin/dokka/releases/tag/v1.7.20) to [1.8.10](https://github.com/Kotlin/dokka/releases/tag/v1.8.10).

## 🔗 Dependencies

This issue is blocked by the following ones:

- [x] #172 
- [x] #261 

## ✅ Checklist

- [x] Document what's new in this version in comment below.
- [x] Upgrade the Gradle plugin and test the documentation generation.
- [x] Refactor the Gradle build script if needed.
- [x] Close this issue as completed and update tracking ones if present.

**Comment by LVMVRQUXL** [comment]
We should move the automation tasks to a new issue.

**Comment by LVMVRQUXL** [comment]
No notable changes introduced in [Dokka 1.8.10] could impact Kotools Types.

[Dokka 1.8.10]: https://github.com/Kotlin/dokka/releases/tag/v1.8.10

**Comment by LVMVRQUXL** [comment]
This upgrade may resolve the following security vulnerability detected from jackson-databind 2.12.7, a transitive dependency of Dokka 1.7.21: [Uncontrolled Resource Consumption](https://github.com/advisories/GHSA-rgv9-q543-rqg4).
